alacritty_terminal/src/config/mod.rs

```diff
@@ -15,37 +15,43 @@ pub use crate::config::scrolling::Scrolling;
 pub const LOG_TARGET_CONFIG: &str = "alacritty_config_derive";
 const MIN_BLINK_INTERVAL: u64 = 10;
 
-pub type MockConfig = Config<HashMap<String, serde_yaml::Value>>;
-
 /// Top-level config type.
 #[derive(ConfigDeserialize, Debug, PartialEq, Default)]
-pub struct Config<T> {
+pub struct Config {
     /// TERM env variable.
     pub env: HashMap<String, String>,
 
     pub selection: Selection,
 
-    /// Path to a shell program to run on startup.
-    pub shell: Option<Program>,
-
     /// How much scrolling history to keep.
     pub scrolling: Scrolling,
 
     /// Cursor configuration.
     pub cursor: Cursor,
 
+    #[config(flatten)]
+    pub pty_config: PtyConfig,
+}
+
+#[derive(ConfigDeserialize, Clone, Debug, PartialEq, Default)]
+pub struct PtyConfig {
+    /// Path to a shell program to run on startup.
+    pub shell: Option<Program>,
+
     /// Shell startup directory.
     pub working_directory: Option<PathBuf>,
 
-    /// Additional configuration options not directly required by the terminal.
-    #[config(flatten)]
-    pub ui_config: T,
-
     /// Remain open after child process exits.
     #[config(skip)]
     pub hold: bool,
 }
 
+impl PtyConfig {
+    pub fn new() -> Self {
+        Default::default()
+    }
+}
```

Comment:
If you implement `From`/`Into` for `TerminalOptions` -> `PtyConfig` you probably don't need `PtyConfig::new` anymore.

Reply:
This is technically unused now, just no warning because it's a public API.

Do you want to keep this around just to make the shorter `::new` available in our public API and make it easier to understand how this can be initialized when looking at rustdoc, or should we remove this?

Reply:
I just like to have `new` constructor in general for such types. We can remove it, but having I don't see a reason why.